fix(amplify-category-custom): validate that the project has been initialized

SebasRod23 opened the pull request:
#### Description of changes

Get meta file to validate that the project has been initialized before running the command

#### Issue #, if available

Closes #8920 

#### Description of how you validated changes

It uses the `pathManager` to validate that the meta file exist, if it doesn't then the it will throw a `NotInitializedError` before trying to run the command. This applies for every command in the `custom` category.
#### Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

SebasRod23 committed:
fix(amplify-category-custom): validate that the project has been initialized before running the command

jhockett commented on packages/amplify-category-custom/src/index.ts:
```diff
@@ -18,6 +18,8 @@ export async function executeAmplifyCommand(context: $TSContext) {
 
   const commandModule = await import(commandPath);
 
+  pathManager.getAmplifyMetaFilePath();
```
This should work, but it's confusing to rely on the error throwing side effect of `pathManager.getAmplifyMetaFilePath();` when we don't need the meta file path.

A better solution would be to add a stateManager method like `isProjectInitialized()` and call that here. That will make it easier to provide a useful error message, instead of throwing an error.

SebasRod23 replied:
Thank you for the feedback. I just have some questions: should I still call `pathManager.getAmplifyMetaFilePath()` in the `isProjectInitialized()` to validate that the project is initialized or is there a better way to do the validation? and, should the new method throw an error or just the error message?

jhockett replied:
On second thought, I think we can just use an existing stateManager method and leave a comment. I think implementing `isProjectInitialized()` robustly is out of scope for this issue.

```ts
// Check if project has been initialized
if (!stateManager.metaFileExists()) {
  printer.error('Could not find the amplfiy-meta.json file. Make sure your project is initialized in the cloud.');
}
```

jhockett reviewed (changes requested):
Hi @SebasRod23, thanks for the contribution! I left a comment about reworking the solution to be more maintainable. Let me know if you are able to implement the changes or if you have any questions.

SebasRod23 committed:
refactor: change validation function and error

Update the validation function from pathManager to stateManager, also and error is no longer thrown
just an error message

Closes #8920
